### ISSUE TITLE
support setting a package root for the analysis server

### ISSUE THREAD

**Issue by devoncarew** [dart-atom/dartlang#635]
@stereotype441

**Comment by johncblandii** [comment]
This would be great. An array would be better so we could say: analyze `lib/`, `web/`, and `test/`. I'm sure others have their desired paths, but this'd be a great start!

**Comment by stereotype441** [comment]
The API for setting this can be found here: https://htmlpreview.github.io/?https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/doc/api.html#request_analysis.setAnalysisRoots

Basically you include a `packageRoots` map in your `analysis.setAnalysisRoots` request mapping from each analysis root to the corresponding package root.  So for example if the user has requested to analyze packages `/foo` and `/bar`, and they want `/foo` to be analyzed with a package root of `/package-root-for-foo`, you would send a request like this:

    {
      "id": ...
      "method": "analysis.setAnalysisRoots"
      "params": {
        "included": ["/foo", "/bar"],
        "excluded": [],
        "packageRoots": {
          "/foo": "/package-root-for-foo"
        }
      }
    }

**Comment by johncblandii** [comment]
I actually just found this on the `analyzer` package. :-D You can set project settings with [.analysis-options`](https://pub.dartlang.org/packages/analyzer).

This ticket can be closed, IMHO.

**Comment by devoncarew** [comment]
Hey @johncblandii, just to clarify, this issue is scoped to a pretty specific feature request for the atom plugin :)

**Comment by johnbland-wf** [comment]
Is it not to target analyzer to certain directories? Are we talking two different things?

**Comment by devoncarew** [comment]
Yup, I think so. `.analysis_options` is super useful, but I think orthogonal to this particular issue.

**Comment by johncblandii** [comment]
Also doesn't work. Seems the plugin has to send those options. It doesn't respect it.
